internal/cloudflare/dns.go

```diff
@@ -0,0 +1,47 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	cloudflarego "github.com/cloudflare/cloudflare-go"
+)
+
+func (c *Cloudflare) NukeDNSRecords(ctx context.Context) error {
+	c.logger.Infof("removing dns records for domain %q", c.zoneName)
+
+	records, _, err := c.client.ListDNSRecords(ctx, &cloudflarego.ResourceContainer{
+		Identifier: c.zoneID,
+	}, cloudflarego.ListDNSRecordsParams{})
+	if err != nil {
+		return fmt.Errorf("unable to list records for domain %q: %w", c.zoneName, err)
+	}
+
+	// if subdomain is set filter the records
+	var subdomainRecords []cloudflarego.DNSRecord
```

Comment:
You can probably preallocate this so `append` doesn't create `len(records)+1` allocations and instead just do one:

```suggestion
	subdomainRecords := make([]cloudflarego.DNSRecord, 0, len(records))
```